tests/lsan.supp

```diff
@@ -23,15 +44,56 @@ leak:_ctypes_alloc_format_string
 leak:__strdup
 
 ####
-#### Miscellaneous 3rd party libraries, test code, etc:
+#### Miscellaneous 3rd party libraries:
 ####
 
-leak:*libpython*
-leak:*libwebsockets*
-leak:*python2*
+### Python
 
-# We should be able to uncomment these once all known dispatch leaks have been fixed
-leak:*libqpid-proton*
+# these Python leaks happen even after simple Py_Initialize(); Py_Finalize();
+#  https://bugs.python.org/issue1635741
+leak:^_PyObject_Realloc
+leak:^PyObject_Malloc$
+leak:^PyThread_allocate_lock$
 
-# Ignore test code
-leak:run_unit_tests.c
+# the PyMalloc mechanism is incompatible with Valgrind, it must be disabled or reported "leaks" must be suppressed
```

Comment:
If SANITIZE_3RD_PARTY is OFF, these following Python leak lines are overridden by leak:/libpython2.*.so\nleak:/libpython3.*.so\n ??

Reply:
Yes, exactly. These following lines require Python debug symbols to be installed, so the function names can be matched.

Reply:
So, if I understand correctly, no matter what you have in this list below, having the leak:/libpython2..so\nleak:/libpython3..so\n at the end of the file totally ignores this list. So, does it make sense to not even have this list if we are going to have the above two lines at the end of the file ?

Reply:
We do not append the `leak:/libpython3..so` always, only when the SANITIZE_3RD_PARTY option is OFF. In GitHub Actions, the Python debug info is installed and works ok, so there SANITIZE_3RD_PARTY=ON and the more granular suppressions are used. (And therefore e.g. the leak in IoAdapter is picked up). Same thing on my laptop.

On Travis, and on other people's computers, the debug symbols might not be present. In that case, the whole-library suppressions are needed, otherwise they would see test failures.

Reply:
It would be great if the following list *and* leak:/libpython2..so\nleak:/libpython3..so\n don't appear at the same time in the file.

Reply:
Why?

The way it is now (in the PR), the `lsan.supp` (or `lsan.supp.in` if I make that change) is a valid and usable LSan suppression file by itself, without any preprocessing. If I to do what you suggest, this will no longer be the case: (it will either contain CMake variable and won't be valid because of it, or it will not contain the individual Python function suppressions and will be incomplete).

Reply:
In case you end up doing the configure_file from the first comment, you could add this as well? It is deceiving to see a  suppression list being overridden by two lines at the bottom.

Reply:
It isn't really overridden. The two .so suppressions simply cover all the Python suppressions that went before. I am thinking I could remove the conditional CMake logic and put the .so suppressions for python there always.

The only leak I flushed out by doing this in the more complicated way is th `IoAdapter` one, and that is fairly insignificant. There is about 4 or 5 instances of the class, only. In the whole qdr run. So that leak is hardly the most significant one.